### ISSUE TITLE
When I frequently send the switch control command for a while, the SDK crashes. Can you help me?

### ISSUE THREAD

**Issue by codepgq** [NordicSemiconductor/IOS-nRF-Mesh-Library#179]
![image](https://user-images.githubusercontent.com/20208598/67372608-bf0f2200-f5b0-11e9-9aaa-a332061089fa.png)

``` swift
    lazy var btnONOFF: PQButton = {
        let view = PQButton()
        //view.backgroundColor = UIColor.gray
        view.setImage(UIImage(named: "button_off"), for: .normal)
        view.setImage(UIImage(named: "button_on"), for: .selected)
        view.buttonClick { [weak self] (button) in
            self?.colorPanelView.isHasOverlay = button.isSelected
            button.isSelected = !button.isSelected
            
            /// send color
            if let model = self?.node.onoffModel {
                let colorMsg = GenericOnOffSet(button.isSelected)
                do {
                    try MeshNetworkManager.instance.send(colorMsg, to: model)
                } catch {
                    print("send  onoff failed", error)
                }
            }
        }
        return view
    }()
```

**Comment by philips77** [comment]
Sure, thank you for submitting the issue. I will look into it soon.

**Comment by philips77** [comment]
Hi, could you try with the latest commit? I added some synchronisation around the contexts, I hope it will help.

**Comment by codepgq** [comment]
I will test it tomorrow

**Comment by philips77** [comment]
I'm planning to release 2.0 today (fingers crossed!). If you'll find any issues, they'll be in 2.0.1+.

**Comment by codepgq** [comment]
OK！

**Comment by philips77** [comment]
Any success here?

**Comment by philips77** [comment]
The 2.0.0 has been released and it's available on CocoaPods and Carthage. As I said, if you'll find any more issues they'll be addresses in future versions.

**Comment by codepgq** [comment]
I'm sorry to have replied to you now, we haven't tested the problem since your last fix. One phenomenon, when I have several model binding applicationkeys, some models may not be bound on, this is not always happening. But after 0.1s each time the data is spaced, it's work better, and I remember there was a message queue on your sdk

**Comment by codepgq** [comment]
hi,I have this problem again now, after I quickly control many times of ctl instructions.

![image](https://user-images.githubusercontent.com/20208598/67915938-31928a00-fbcf-11e9-8ec9-8daa7404fe05.png)


![image](https://user-images.githubusercontent.com/20208598/67915830-e7110d80-fbce-11e9-8636-614229cab269.png)

**Comment by philips77** [comment]
It's one layer below. Looks like the same issue with concurrent modification, isn't it?

**Comment by philips77** [comment]
I added thread synchronisation on Upper Transport Layer. I'll release it in 2.0.1 on Monday. If you could try the *develop* branch and see if it works for you?

**Comment by codepgq** [comment]
I'll be testing it next Monday, and now I don't have the device?.

**Comment by philips77** [comment]
Fix has been released in 2.1.0. I'm closing the issue now, feel free to reopen it or create a new one if you still hit this issue.
